### PR TITLE
raftstore-v2: store heartbeat add kv size and snap size

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What is changed and how it works?

Issue Number: Ref https://github.com/tikv/tikv/issues/12842

What's Changed:

```commit-message
1. store heartbeat should add snapshot and kv engine used size 
```

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 

```release-note
 None.

```

### CODE REVIEW

File: components/raftstore-v2/src/operation/pd.rs [hunk 1]
```diff
@@ -53,7 +53,8 @@ impl Store {
 
         // stats.set_query_stats(query_stats);
 
-        let task = pd::Task::StoreHeartbeat { stats };
+        let snap_size = ctx.snap_mgr.total_snap_size().unwrap();
```
Comment (tabokie) [inline components/raftstore-v2/src/operation/pd.rs:56]: `total_snap_size` is heavy. Clone the `snap_mgr` to pd runner instead.

Comment (bufferflies) [reply components/raftstore-v2/src/operation/pd.rs]: fixed

File: components/raftstore-v2/src/worker/pd/store_heartbeat.rs [hunk 2]
```diff
@@ -257,7 +257,6 @@ where
         self.remote.spawn(f);
     }
 
-    /// Returns (capacity, used, available).
```
Comment (tabokie) [inline components/raftstore-v2/src/worker/pd/store_heartbeat.rs:260]: Why remove it?

**Review by tabokie — APPROVED** [review]

File: components/raftstore-v2/src/worker/pd/store_heartbeat.rs (hunk 2 above)
Comment (bufferflies) [reply components/raftstore-v2/src/worker/pd/store_heartbeat.rs]: fixed

File: components/raftstore-v2/src/operation/pd.rs [hunk 3]
```diff
@@ -50,9 +50,7 @@ impl Store {
         stats.set_bytes_written(0);
         stats.set_keys_written(0);
         stats.set_is_busy(false);
-
-        // stats.set_query_stats(query_stats);
-
+        // todo: add query stats
```
Comment (iosmanthus) [inline components/raftstore-v2/src/operation/pd.rs:53]:
```suggestion
        // TODO: add query stats
```

**Review by iosmanthus — APPROVED** [review]

File: components/raftstore-v2/src/operation/pd.rs (hunk 3 above)
Comment (bufferflies) [reply components/raftstore-v2/src/operation/pd.rs]: fixed

**Review by tonyxuqqi — APPROVED** [review]

**Pull request merged**